Sources/BuildServerProtocol/BuildTargets.swift

```diff
@@ -0,0 +1,119 @@
+import LanguageServerProtocol
+
+/// The workspace build targets request is sent from the client
+/// to the server to ask for the list of all available build
+/// targets in the workspace.
+public struct BuildTargets: RequestType, Hashable {
+  public static let method: String = "workspace/buildTargets"
+  public typealias Response = BuildTargetsResult
+
+  public init() {}
+}
+
+public struct BuildTargetsResult: ResponseType, Hashable {
+  public var targets: [BuildTarget]
+}
+
+public struct BuildTarget: Codable, Hashable {
+  /// The target’s unique identifier 
+  public var id: BuildTargetIdentifier
+
+  /// A human readable name for this target.
+  /// May be presented in the user interface.
+  /// Should be unique if possible.
+  /// The id.uri is used if None. 
+  public var displayName: String?
+
+  /// The directory where this target belongs to. Multiple build targets are allowed to map
+  /// to the same base directory, and a build target is not required to have a base directory.
+  /// A base directory does not determine the sources of a target, see buildTarget/sources. 
+  public var baseDirectory: URL?
+
+  /// Free-form string tags to categorize or label this build target.
+  /// For example, can be used by the client to:
+  /// - customize how the target should be translated into the client's project model.
+  /// - group together different but related targets in the user interface.
+  /// - display icons or colors in the user interface.
+  /// Pre-defined tags are listed in `BuildTargetTag` but clients and servers
+  /// are free to define new tags for custom purposes.
+  public var tags: [BuildTargetTag]
+
+  /// The capabilities of this build target. 
+  public var capabilities: BuildTargetCapabilities
+
+  /// The set of languages that this target contains.
+  /// The ID string for each language is defined in the LSP. 
+  public var languageIds: [String]
+
+  /// The direct upstream build target dependencies of this build target 
+  public var dependencies: [BuildTargetIdentifier]
+
+  public init(id: BuildTargetIdentifier,
+              displayName: String?,
+              baseDirectory: URL?,
+              tags: [BuildTargetTag],
+              capabilities: BuildTargetCapabilities,
+              languageIds: [String],
+              dependencies: [BuildTargetIdentifier]) {
+    self.id = id
+    self.displayName = displayName
+    self.baseDirectory = baseDirectory
+    self.tags = tags
+    self.capabilities = capabilities
+    self.languageIds = languageIds
+    self.dependencies = dependencies
+  }
+}
+
+public struct BuildTargetIdentifier: Codable, Hashable {
+  public var uri: URL
+
+  public init(uri: URL) {
+    self.uri = uri
+  }
+}
+
+public enum BuildTargetTag: String, Codable {
```

Comment:
This should use a `struct` with `RawRepresentable`, since it's an open set.  I.e.
```swift
public struct BuildTargetTag: Codable, RawRepresentable {
  public var rawValue: String
  public init(rawValue: String) {
    self.rawValue = rawValue
  }

  // Not sure if 'Self' actually works here in the current compiler, but you get the idea.
  public static let library: Self = Self(rawValue: "library")
  ...
}
```